packages/builders/src/components/Assertions.ts

```diff
@@ -1,38 +1,60 @@
 import { s } from '@sapphire/shapeshift';
 import { APIMessageComponentEmoji, ButtonStyle } from 'discord-api-types/v10';
-import type { SelectMenuOptionBuilder } from './selectMenu/SelectMenuOption';
-import { UnsafeSelectMenuOptionBuilder } from './selectMenu/UnsafeSelectMenuOption';
+import { SelectMenuOptionBuilder } from './selectMenu/SelectMenuOption';
+import { isValidationEnabled } from '../util/validation';
 
-export const customIdValidator = s.string.lengthGreaterThanOrEqual(1).lengthLessThanOrEqual(100);
+export const customIdValidator = s.string
+	.lengthGreaterThanOrEqual(1)
+	.lengthLessThanOrEqual(100)
+	.setValidationEnabled(isValidationEnabled);
 
-export const emojiValidator = s.object({
-	id: s.string,
-	name: s.string,
-	animated: s.boolean,
-}).partial.strict;
+export const emojiValidator = s
+	.object({
+		id: s.string.setValidationEnabled(isValidationEnabled),
+		name: s.string.setValidationEnabled(isValidationEnabled),
+		animated: s.boolean.setValidationEnabled(isValidationEnabled),
```

Comment:
No point in setting these predicate's validation enabled, as the basic typeof checks will still run regardless of the setting